### ISSUE TITLE
Provide (and require) block icon component to ensure accessibility attributes are present

### ISSUE THREAD

**Issue by tofumatt** [WordPress/gutenberg#9372]
Related to #9269 and #9299, specifically: https://github.com/WordPress/gutenberg/issues/9269#issuecomment-415397686

We should create a component that allows developers to supply an SVG or acts like an SVG, but adds required accessibility attributes to it like in #9299. We should be using it (to set an example and avoid regressions like in #9269.

Ideally we could deprecate icon props that don't use our new `<BlockIcon>` component or whatever, so we can have more control over how they're rendered in the future.
